src/main/webapp/script.js

```diff
@@ -21,9 +21,27 @@ function getUser() {
 }
 
 function searchEvents() {
+  document.getElementById('events-container').innerText = "";
+  var search = document.getElementById('search').value;
+  var category = document.getElementById('category').value;
+  var duration = document.getElementById('duration').value;
+  var location = document.getElementById('location').value;
   fetch('/events?' + new URLSearchParams({
     search: search,
-  })).then(response => response.json());
+}) + '&' + new URLSearchParams({
+    category: category,
+}) + '&' + new URLSearchParams({
+    duration: duration,
+}) + '&' + new URLSearchParams({
+    location: location,
+})).then(response => response.json()).then(events => events.forEach(showEvent));
+}
+
+function showEvent(event) {
+  document.getElementById('events-container').innerHTML += "<div><h1>" 
```

Comment:
Maybe not in this PR, but maybe it's nicer to create elements with document.createElement(tagName)

Reply:
I will do that in the next one.